### PR TITLE
[APB-3446][MP,PT] Add page for check agent details, reformat with scalafmt

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/uk/gov/hmrc/agentstatuschangefrontend/journeys/AgentStatusChangeJourneyModel.scala [hunk 1]
```diff
@@ -28,11 +33,45 @@ object AgentStatusChangeJourneyModel extends JourneyModel {
   object State {
 
     case object Search extends State
+    case class DetailsArn(arn: Arn, agentName: String, agentAddress: Address, status: String) extends State
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentstatuschangefrontend/journeys/AgentStatusChangeJourneyModel.scala]: Just a thought for future.. could have a single Details state with an `Either[Arn, Utr]` in it. 
Might help reduce some duplication of handling code.

Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentstatuschangefrontend/journeys/AgentStatusChangeJourneyModel.scala]: Yeah i suppose thats more of a design decision as to how specific you want to make the states

File: app/uk/gov/hmrc/agentstatuschangefrontend/views/details.scala.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+@(pageConfig: DetailsPageConfig)(implicit request: Request[_], messages: Messages, config: Configuration)
+
+@mainTemplate(title = Messages("details.title"), bodyClasses = None) {
+
+    @back_link(pageConfig.backLink)
+
+    <h1 class="heading-xlarge margin-bottom-30 margin-top-0">@Messages("details.title")</h1>
+
+    @if(pageConfig.agentIdentifierType == "arn"){
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentstatuschangefrontend/views/details.scala.html]: Just a thought. Could pass the `agentIdentifier` as some subtype of `TaxIdentifier`, then safely pattern match here on the type here instead. Then be liberated of the special `agentIdentifierType` string values

File: app/uk/gov/hmrc/agentstatuschangefrontend/views/details.scala.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    @for(addressLine <- pageConfig.agentAddress.lines) {
+        <p class="margin-top-0 margin-bottom-0">@addressLine</p>
+    }
+    <p>@pageConfig.agentAddress.postcode.getOrElse("")</p>
+
+    <div class="panel panel-border-wide">
+        <p>@pageConfig.status</p>
+    </div>
+
+    @if(pageConfig.status == "suspended") {
```
Comment (johnrevill) [inline app/uk/gov/hmrc/agentstatuschangefrontend/views/details.scala.html]: Might want to raise the status from a String to a type of its own such as a sealed trait. That'd prevent special magic String values propagating throughout the codebase and keep us type safe.

File: test/journey/AgentStatusChangeJourneyModelSpec.scala [hunk 4]
```diff
@@ -42,9 +42,34 @@ class AgentStatusChangeJourneyModelSpec
...
+      }
+      "transition to DetailsUtr when Search submitted a form with a valid Utr" in {
+        given(Search) when submitSearch(getAgentDetails)("strideId")("3110118001") should thenGo(
+          DetailsUtr(
+            Utr("3110118001"),
+            "agent name",
+            Address(Seq("1 street", "1 town"), Some("POSTCODE")),
+            "suspended"))
+      }
+      "transition to Search when there are no agent detials found" in {
```
Comment (johnrevill) [inline test/journey/AgentStatusChangeJourneyModelSpec.scala]: Pedantic: details

**Pull request merged**